### DISCUSSION TITLE
Cannot define custom elements

### DISCUSSION THREAD

**Discussion by BTrifonov** [clientIO/joint#2304] [Q&A] [answered]
### Introduction

Hi, 
I am trying to create a custom element with the define() method, however when I instantiate the element I get the warning that my element cannot be found in jointjs: "export 'shapes'.'custom' (imported as 'joint') was not found in 'jointjs'".
My define method: joint.shapes.standard.Circle.define('custom.place',...)
My instantiation: new joint.shapes.custom.place(...)

P.S I am working on a vue project and have imported jointjs like this in each component: "import * as joint from 'jointjs'". So far I was able create a plane and draw different elements. 

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by jamesgeorgewilliams** [comment, marked as answer]
You can read more about custom elements & `define` at the following links:
https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Cell.define
https://resources.jointjs.com/tutorial/custom-elements

You can do the following:
```js
joint.dia.Element.define('custom.place', {}, {});
```

If you are using Vue, you can look at the different branches of the following repository to get some inspiration about project structure.
https://github.com/clientIO/joint-plus-tutorial-vue
https://resources.jointjs.com/tutorial/vue-ts

The following custom shapes tutorial might be useful for you too:
https://resources.jointjs.com/tutorial/ts-shape

**Reply by BTrifonov** [reply]
Thank you!
